src/Resource.lua

```diff
@@ -9,15 +9,51 @@ local Resource = {
         fuel = getObjectFromGUID(resources_GUID["fuel"]),
         material = getObjectFromGUID(resources_GUID["materials"])
 
+    },
+    clusters = {
+        {
+            ["a"] = "weapon", 
+            ["b"] = "fuel", 
+            ["c"] = "material"
+        },
+        {
+            ["a"] = "psionic", 
+            ["b"] = "weapon", 
+            ["c"] = "relic"
+        },
+        {
+            ["a"] = "material", 
+            ["b"] = "fuel", 
+            ["c"] = "weapon"
+        },
+        {
+            ["a"] = "relic", 
+            ["b"] = "fuel", 
+            ["c"] = "material"
+        },
+        {
+            ["a"] = "weapon", 
+            ["b"] = "relic", 
+            ["c"] = "psionic"
+        },
+        {
+            ["a"] = "material", 
+            ["b"] = "fuel", 
+            ["c"] = "psionic"
+        }
     }
 }
 
 function Resource:take(name, pos)
     LOG.DEBUG("name:" .. name)
-    self.supplies[name].takeObject({
+    return self.supplies[name].takeObject({
         position = pos,
         smooth = true
     })
 end
 
+function Resource:getSystem(cluster, system)
```

Comment:
A more appropriate name for this function would be Resource:get_from_system(cluster, system) or maybe just Resource:from_system(cluster, system) or Resource:from(cluster, system)

Reply:
Actually, it returns a resource name and not a Resource object. It should be named Resource:name_from_system or Resource:name_from or Resource:name